### PR TITLE
12 lint and prettier

### LINKED ISSUE #12: ci: run lint and prettier before deploying

### PR DESCRIPTION
**Pull request by Lukasd**
closes #12

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  echo 'Make changes required above and try again.'
+  false;
+)
+
+# Everything went well so far, lets build
+npm run build ||
+(
+  echo 'Next build failed. See the errors above.'
+  false;
+)
```
Comment (saihaj) [inline .husky/pre-commit:32]: I think we don't need to build it on every commit cause that can slow things down significantly.

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  },
+  plugins: ['react', '@typescript-eslint'],
+  settings: {
+    react: {
+      version: 'detect'
+    }
+  },
+  rules: {
+    'no-use-before-define': 'off'
+  }
```
Comment (saihaj) [inline .eslintrc.js:23]: with React 17 we don't need need to import `React` https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#nextjs so we should add these 2 rules
```js
    "react/jsx-uses-react": "off",
    "react/react-in-jsx-scope": "off",
```

**Review by saihaj — CHANGES_REQUESTED** [review]

File: .husky/pre-commit (hunk 1 above)
Comment (Lukasd) [reply .husky/pre-commit:32]: That's true, I'll remove it.

File: .eslintrc.js (hunk 2 above)
Comment (Lukasd) [reply .eslintrc.js:23]: Awesome, didn't know that. I'll update it.

File: tsconfig.json [hunk 3]
```diff
@@ -14,6 +14,11 @@
     "isolatedModules": true,
     "jsx": "preserve"
   },
-  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx"],
+  "include": [
+    "next-env.d.ts",
+    "**/*.ts",
+    "**/*.tsx",
+    "src/pages/_document.tsx"
```
Comment (saihaj) [inline tsconfig.json:21]: don't then we need this since above 2 include everything.

Comment (Lukasd) [reply tsconfig.json:21]: I can remove the `"src/pages/_document.tsx"`, but the other ones get added back on `next build` anyway, or I could leave an empty array and disable the lint rule that checks it.

**Review by saihaj — APPROVED** [review]

**Comment by saihaj** [comment]
Awesome 👏🏼

**Pull request merged**